cpp/DIS/VariableDatum.cpp

```diff
@@ -44,56 +38,64 @@ void VariableDatum::setVariableDatumLength(unsigned int pX)
     _variableDatumLength = pX;
 }
 
-char* VariableDatum::getVariableDatums()
+char* VariableDatum::getVariableDatums() // a bit dangerous. we could just return a ref to _variableDatums
 {
-    return _variableDatums;
+    return _variableDatums.data();
 }
 
 const char* VariableDatum::getVariableDatums() const
 {
-    return _variableDatums;
+    return _variableDatums.data();
 }
 
-void VariableDatum::setVariableDatums(const char* x, int length)
+void VariableDatum::setVariableDatums(const char* x, const int length)
 {
-	// Trying to set something too big. Punt, for now.
-	if(length > STATIC_ARRAY_LENGTH)
-	{
-		std::cout << " The VariableDatum is too large to fit into the VariableDatum object. Punting." << std::endl;
-		return;
-	}
-
-	int byteLength = length;
-	_variableDatumLength = length * 8; // in bits
+    // Should we still check for too large a size? This particular limitation no longer holds.
```

Comment:
Ya now that we are resizing the datum below, I'd be Ok if we removed these size checks here.